### PR TITLE
Configuration for making jest work with React components

### PR DESCRIPTION
**Pull request by Eddges**
@sarathms kindly review. I hope to work with this configuration during the initial weeks of the coding phase.

Also, I wasn't able to make the **.test.js and **.spec.js files run before the e2e tests. Do you have any suggestions on how it can be done in /test/lib/setup.js file? There seems to be sort function with a link in the comments for reference, but the link doesn't work.

### CODE REVIEW

**Review by sarathms — CHANGES_REQUESTED** [review]
This is a great start @Eddges. I have let a few comments requesting some improvements. Primarily, the suggestion is to separate out unit tests and e2e tests which eliminates the question you asked about sequencing them.

The link I was referencing is https://github.com/facebook/jest/issues/6194#issuecomment-521521514 It does behave weird for some reason.

File: setupTests.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import "jest-canvas-mock";
```
[the remark this answers is not in the document]
Comment (Eddges) [reply setupTests.js:1]: This is for the `react-lottie-player` library

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["next/babel"]
```
[the remark this answers is not in the document]
Comment (Eddges) [reply .babelrc:2]: I think `.babelrc` is supposed to be in the directory same as `package.json` (root directory) for translating the ES6 syntax, as the mentioned syntax was not supported earlier.
I'm not sure what purpose the file in `renderer` directory is serving. I deleted the file from the `renderer` directory and couldn't notice any difference. On building the app for development, the process seemed to utilise the `.babelrc` in root directory.

File: test/demo.test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+const renderWithReactIntl = (component, locale, messages) => {
+    return render(
+        <IntlProvider locale={locale} messages={messages}>
+            { component }
+        </IntlProvider>
+    );
+};
+
+const ThemedComponent = ({ Component }) => {
```
[the remark this answers is not in the document]
Comment (Eddges) [reply test/demo.test.js]: Sounds like a better approach, thanks! I have tried to implement it in the newer commit, kindly check.

File: test/lib/sequencer.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ class CustomSequencer extends TestSequencer {
     // The listed test files will be processed in this order
     // Other `*.e2e.js` files will be picked after these in no specific order
     const orderPath = [
+      'demo',
```
[the remark this answers is not in the document]
Comment (Eddges) [reply test/lib/sequencer.js]: Done!

File: .babelrc (hunk 2 above)
[the remark this answers is not in the document]
Comment (sarathms) [reply .babelrc:2]: Ah right! This for ES6+ syntax support in the test code. Makes sense now. In which case, we can just move `renderer/.babelrc` to the root of the repo.

If at some point, a common configuration doesn't work, we can break it up as needed.

File: jest.config.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    testPathIgnorePatterns: ["/.next/", "/node_modules/"],
+    setupFilesAfterEnv: ["./setupTests.js"],
+    verbose: true,
+    roots: ["<rootDir>/test"],
+    testMatch: ["**/test/**/*.e2e.js", "**/test/**/*.test.js"],
```
[the remark this answers is not in the document]
Comment (sarathms) [reply jest.config.js]: Fair enough. This new setup is good enough. We can optimize for better readability eventually.

File: jest.config.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+    testPathIgnorePatterns: ["/.next/", "/node_modules/"],
+    setupFilesAfterEnv: ["./setupTests.js"],
```
Comment (sarathms) [inline jest.config.js:3]: Can we move this into `test/lib/setup.js` to keep all tests related code under `test`? The `jest.Timeout` directive in that file currently does not have any impact, so you can remove it. It was added long back when tests were running in dev environment.

Comment (sarathms) [inline jest.config.js:3]: Can you move this into `test/lib/setup.js` to keep all tests related code under `test/`? The `jest.Timeout` directive in that file currently does not have any impact, so you can remove it. It was added long back when tests were running in dev environment.

**Review by sarathms — APPROVED** [review]
Looks good to me! 👍 

I have left a couple of suggestions for minor tweaks. Apart from those, can you also add a new step in the CI script for unit testing that runs `yarn test:unit` before the integration tests?

Once these are done and the CI scripts runs well, I can merge this new setup for us to build upon.

File: jest.config.js (hunk 6 above)
Comment (Eddges) [reply jest.config.js]: Okay. I did move all the files under the `test` directory. I have renamed `setup.js` to `jest.config.js` if that's alright.

File: renderer/components/onboard/__tests__/Sections.test.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+             <ThemeProvider theme={theme}>
+                 { component }
+             </ThemeProvider>
+         </IntlProvider>
+     );
+ };
+ 
+ describe("Rendering Sections", () => {
+     test("Sections is correctly mounted", async () => {
+         renderComponent(<Sections />, "en", English);
```
[the remark this answers is not in the document]
Comment (Eddges) [reply renderer/components/onboard/__tests__/Sections.test.js]: Done, thanks! This seems much better.

**Pull request merged**